### ISSUE TITLE
AI sched 61c: sibling-repo SLM-OS trace ingester + fine-tuning mode

### ISSUE THREAD

**Issue by johnjezl** [SLM-OS/SLM-Operating-System#879]
## Description

Sub-ticket of #61. **Work happens in the sibling repo `~/projects/slm-os-scheduler-ai`, not in this repo.** Filed here for capstone tracking.

Add a SLM-OS-trace data loader to the training pipeline. Add a fine-tuning mode that starts from the existing synthetic-trained MLP/PPO and adapts on captured SLM-OS traces rather than training from scratch.

Depends on #61b (#58 / #61b trace-format ships first) being merged in the SLM-OS repo so the ingester has a stable format target.

## Why fine-tune, not retrain from scratch

A 5-second SLM-OS trace at scheduler tick rate (~100 Hz) yields ~500 state-vector samples. A 30-minute capture yields ~180K. The MLP has 131K parameters; PPO is similar. Training from scratch on this volume risks overfitting and producing worse weights than the synthetic-trained baseline. Fine-tuning from pretrained weights with a low learning rate is the correct shape — matches how embedded-ML pipelines adapt models to real-world data in production.

## Scope (sibling repo)

1. **New trace ingester** at `~/projects/slm-os-scheduler-ai/data/slmos_traces.py`:
   - Parses the binary format defined in #61b's `docs/sched-trace-format.md`.
   - Validates magic, version, checksum.
   - Reshapes records to match the existing simulator observation schema (108-dim state, action triple).
   - Synthesizes a "reward" signal from the outcome window (high reward = met-deadline + low completion-time; low reward = missed-deadline). Reuse the simulator's existing reward function from `slm_sim/reward.py` so the loss surface is consistent.
   - Outputs a Parquet file in the same shape as `slm_sim`'s training data.
2. **Add `--source slmos-traces` flag** to `scripts/_train_mlp.py` and `scripts/_train_ppo.py`:
   - When set, skip simulator dataset generation; load from the Parquet emitted by the ingester.
   - Use a lower learning rate (e.g. 0.1× the from-scratch rate) and fewer epochs (e.g. 5-10) — these are fine-tuning hyperparameters, document them.
   - Start from the existing checkpoint at `models/mlp/best_jetson_orin_nano.pt` (or PPO equivalent), not from random init.
3. **Output weights with `_real` suffix.** Don't overwrite the synthetic-trained weights. Both should remain available as alternative weight sets per the pluggable-policy framing (#848). The export pipeline (`scripts/export_models.py`) gets a parallel `_real` output path: `ai_weights_mlp_real.c` etc.
4. **Verification.** Run the sibling repo's existing `run_evaluation.py` on the fine-tuned weights using the same scenarios as the synthetic baseline. Capture the comparison numbers.

## Compute requirements

Fine-tuning at this scale should be feasible on the developer's GPU or CPU in well under an hour. Document the measured runtime.

## Acceptance criteria

- `python scripts/_train_mlp.py --source slmos-traces --input trace_001.bin --output models/mlp/best_jetson_orin_nano_real.pt` runs to completion.
- `python scripts/export_models.py --model mlp --platform jetson_orin_nano --weights-checkpoint models/mlp/best_jetson_orin_nano_real.pt` emits `ai_weights_mlp_real.c` alongside the existing `ai_weights_mlp.c`.
- `run_evaluation.py` on the fine-tuned weights produces results comparable in scale to the synthetic baseline (i.e. not catastrophically worse — fine-tune actually preserved generalization).
- Documentation: short README addition in sibling repo describing the SLM-OS-trace flow.

## Out of scope

- SLM-OS-side trace collection → #61b (prerequisite).
- Re-importing and validating retrained weights in SLM-OS → #61d.
- Cache_pressure distribution-shift handling — if #60d / #872 lands, the trace will already include real cache_pressure values; the fine-tune will absorb this naturally. No special handling needed.

## Estimated Effort

~3 days of work in sibling repo + retraining compute time (well under an hour).

## References

- Parent: #61. Prerequisite: #61b (trace-format ships first).
- Sibling repo training scripts: `~/projects/slm-os-scheduler-ai/scripts/_train_mlp.py`, `_train_ppo.py`.
- Sibling repo export pipeline: `~/projects/slm-os-scheduler-ai/scripts/export_models.py`.
- Simulator reward function: `~/projects/slm-os-scheduler-ai/slm_sim/reward.py`.

**Comment by johnjezl** [comment]
Closed by sibling-repo PR [SLM-OS/slm-os-scheduler-ai#3](https://github.com/SLM-OS/slm-os-scheduler-ai/pull/3) (squash commit `3e899f2b`). The ingester + fine-tune flow is now live in the sibling repo; SLM-OS-side wiring lands via #884 (PR #933, also merged).
